SalesforceCommerceCloud/pwa-kit
# W-14867867 Fix invalid query params warnings and allow custom query

# Description
With the change from isomorphic that allow passing custom query (c_${string}). We'd want the commerce-react-sdk also adopt the changes. Commerce-react-sdk hooks will automatically accept any custom query key since we reuse type from isomorphic. However, we need to adjust our hooks to allow custom query key in react-query 

This PR
1. Fixes invalid params warnings seen on the retail react app. This is partially because commerce-react-sdk passes in invalid params from client config into isomorphic funcs
2. Allow custom query key to be present in query key in react query cache



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out code
- npm ci
- Run retail app
- Observe there is not more invalid params warning in the console from any page
- add custom query in any hook calls. E.g product-detail page
```
const {
        data: product,
        isLoading: isProductLoading,
        isError: isProductError,
        error: productError
    } = useProduct(
        {
            parameters: {
                id: urlParams.get('pid') || productId,
                allImages: true,
                c_customQuery: [true, false] => add it here
            }
        },
        {
            // When shoppers select a different variant (and the app fetches the new data),
            // the old data is still rendered (and not the skeletons).
            keepPreviousData: true
        }
    )
```
- Check the api calls, and see that custom query is present in payload. e.g

![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/9fa81dbf-15bf-4274-853a-b5c4c8cdc5f9)

- Check that custom query params is also present in react query cache. e.g 
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/186305a0-6deb-46e6-b97f-544f9a7daf67)

- Add an custom query that does not start with c_. You should see a warning in the console. E.g

![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/44dd3e1c-302b-4ff8-bf87-dbfa3b625d89)

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
